packages/kbn-es/src/cluster.ts

```diff
@@ -437,17 +438,20 @@ export class Cluster {
         });
       }
 
-      // once the cluster is ready setup the native realm
-      if (!skipNativeRealmSetup) {
+      // once the cluster is ready setup the realm
+      if (!skipSecuritySetup) {
         const nativeRealm = new NativeRealm({
           log: this.log,
           elasticPassword: options.password,
           client,
         });
 
         await nativeRealm.setPasswords(options);
-      }
 
+        if (args.some((arg) => arg.includes(MOCK_IDP_REALM_NAME))) {
```

Comment:
__optional nit:__ maybe it'd be safer to do something like this to make sure ES is indeed configured with the SAML realm with this name and not some other realm (it's unlikely to happen in practice, but just in case):
```suggestion
        const samlRealmConfigPrefix = `authc.realms.saml.${MOCK_IDP_REALM_NAME}.`;
        if (args.some((arg) => arg.includes(samlRealmConfigPrefix))) {
```